### ISSUE TITLE
aws_api_gateway_authorizer doesn't support import

### ISSUE THREAD

**Issue by eburcat** [terraform-providers/terraform-provider-aws#9529]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### New or Affected Resource(s)

* aws_XXXXX

### Potential Terraform Configuration

```hcl
# Copy-paste your Terraform configurations here - for large Terraform configs,
# please use a service like Dropbox and share a link to the ZIP file. For
# security, you can also encrypt the files using our GPG public key.
```

### References

* #0000

**Comment by DrFaust92** [comment]
i tried to take a look at this but i think its blocked by the fact that the apigateway authorizer struct does not have an `restapiid` field and when importing an error occurs that the restapiid is empty

i opened a ticket on the aws go sdk repo: https://github.com/aws/aws-sdk-go/issues/2890

hope this is helpful

**Comment by DrFaust92** [comment]
Opened PR #11436 to handle this

**Comment by bflad** [comment]
Support for `aws_api_gateway_authorizer` resource import has been merged and will release with version 2.44.0 of the Terraform AWS Provider, Thursday next week. Thanks to @DrFaust92 for the implementation. 👍
